### ISSUE TITLE
TADProc - explicit error handling related to the missing environment variables

### ISSUE THREAD

**Issue by Sandy-at-Tazama** [frmscoe/transaction-aggregation-decisioning-processor#162]
## User Story
**As a** new operator
**I want** the system to be set up correctly to save evaluation results to the database and
**I want** proper error handling in place should the evaluation database not exist and
**I want** the system to be set up to use environment variables

## Acceptance Criteria
1. TADProc environment variables should include the transaction collection in the evaluationResult database
2. If the transaction collection/evaluationResult database are missing from environment variables in the TADProc startup sequence, start up should fail with a fatal error.
3. The report out of TADProc should always be saved in the database as specified in the TADProc environment variables
4. Remove the optional functionality to save the report in the transactionHistory database
5. Update the deployment guide to co-ordinate/ align the environment variables in TADProc and admin-service API
6. Include error handling for a case where the transaction collection in the evaluationResult database does not exist

## Back story
If the transactions collection in the evaluationResults database does not exist, this should raise a fatal error out of the TADProc.

That collection and database must be defined in the environment variables and if they are missing, the TADProc startup sequence should also fail with a fatal error.

In short, there must always be a evaluationResults.transactions DB (as per the environment variables) and the results must go into this DB. If we are going to retrieve the evaluation results, it should come from this DB. The address of this DB should then also be in an environment variable for the new API and it must be coordinated with the TADProc environment variables (manually, during deployment, I guess).

![image](https://github.com/frmscoe/transaction-aggregation-decisioning-processor/assets/142333955/2cea46f9-913e-42a2-ae76-2f8c9ba783d5)

**Comment by Sandy-at-Tazama** [comment]
We need a way to test this via a GitHub action.  Requirement is:
On PR request create
1.  run a GitHub action to deploy the processor with the recommended .env.  All must pass.
2.  run a GitHub action to deploy the processor with the defaultable/ optional .env variables missing. All must pass.
3. run a GitHub action to deploy the processor with the non-defaultable/ mandatory .env variables missing. None must pass - with error messages.
